### DISCUSSION TITLE
httpresponse body 확인 방법 문의

### DISCUSSION THREAD

**Discussion by kimsw61** [naver/ngrinder#859] [General]
안녕하세요.
로그인한 jwt 토큰의 데이터를 바꿔서 다시 request를 해야하는 문제가 있습니다.

grinder.logger.info("1. login start")
try {
	HTTPResponse response = request.POST(appUrl+"/user-service/v1/signin", body.getBytes())

	if (response.statusCode == 301 || response.statusCode == 302) {
		grinder.logger.warn("Warning. The response may not be correct. The response code was {}.", response.statusCode)
	} else {
		if (response.statusCode != 200) {
			assertThat(response.statusCode, is(200))
		}
	}
	def toJSON = { new JsonSlurper().parseText(response.getBodyText()) }
	grinder.logger.warn(toJSON.toString())
	grinder.logger.warn("1. login end")
} catch (e) {
	grinder.logger.warn(e.toString())
}

이런 형태로 코드를 짯는데 로그인은 정상적으로 처리가 되었지만, 
body 데이터를 확인 할 수 없어서 parse가 되는건지 알수가 없는 상태입니다.

body data를 확인하는 방법과 body 데이터를 parse할 수 있는 방법이 있다면 알려주세요.

감사합니다.

**Comment by imbyungjun** [comment]
안녕하세요.
grinder logger가 오래된 구현체라 출력하는 결과가 약간 혼란스러울 때가 있는데요, 경험상 단순 문자열 타입이 아닌 경우 `grinder.logger.info("{}", someObject)`와 같이 출력하는게 더 알아보기 편했던 것 같습니다.

아래와 같은 방법으로 원하시는 동작을 수행할 수 있을 것 같습니다.

1. 응답 body를 문자열 타입으로 확인하는 방법
```groovy
grinder.logger.info(response.getBodyText())
```
2. JSON 형태의 문자열을 Map으로 변환하는법
```groovy
grinder.logger.info("{}", new JsonSlurper().parseText("{\"key\":\"value\"}"))
```
3. JSON 응답 body를 파싱하여 출력하는 방법 (위와 동일하지만 간단하게 표현하기 위해 `toJson()` 클로저 선언)
```groovy
import groovy.json.JsonSlurper

def toJSON = { new JsonSlurper().parseText(it) }
...
grinder.logger.info("{}", toJson(response.getBodyText()))
```
